src/HealthChecks.System/FolderHealthCheck.cs

```diff
@@ -0,0 +1,41 @@
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace HealthChecks.System
+{
+    public class FolderHealthCheck : IHealthCheck
+    {
+        private readonly FolderHealthCheckOptions _folderOptions;
+
+        public FolderHealthCheck(FolderHealthCheckOptions folderOptions)
+        {
+            _folderOptions = folderOptions ?? throw new ArgumentNullException(nameof(folderOptions));
+        }
+
+        public Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                List<string>? errorList = null;
```

Comment:
I think you copy-pasted initial code from `DiskStorageHealthCheck`. Now it is a bit overcomplicated since your check works only for single folder. I'm OK to have check with multiple folders in options so all that jumping around lists will look normal. Otherwize please remove list.

Reply:
I like the idea of allowing multiple folders, adding that in